server/server/views/event.py

```diff
@@ -91,19 +111,97 @@ def collection_post(self):
-        response = {
-            'event': {},
-        }
-        event_obj = Event.get_event_obj(request,
-                                        id=request.matchdict['event_id'])
+
+        event_obj = Event.get_event_by_id(request, self.event_id)
         if event_obj is None:
             return HTTPNotFound()
+
         tags = EventTag.get_event_tags(request, event_obj.id)
         response = {
             'event': model_to_dict(event_obj),
             'category': event_obj.category.category,
             'tags': tags,
         }
+        if request.user is not None and event_obj.author_id == request.user.id:
+            response['status'] = model_to_dict(self.event_history)
+
+        return response
+
+    @view(schema=EventSchema(), validators=(colander_body_validator,),
+          permission='update')
+    def put(self):
+        """Get validated data and update event data"""
+        request = self.request
+        response = {
+            'success': False,
+            'errors': [{
+                'name': 'name',
+                'description': 'Event name already exist'
+            }],
+        }
+        current_event = Event.get_event_by_id(request, self.event_id)
+
+        if current_event is None:
+            raise HTTPNotFound()
+
+        if (current_event.name == request.validated['name'] or
+            Event.get_event_obj(request,
+                                name=request.validated['name']) is None):
+            event_dict = model_to_dict(current_event)
+            for key in event_dict:
+                if key in request.validated:
+                    event_dict[key] = request.validated[key]
+
+            category_obj = Category\
+                .get_by_name(request, request.validated['category'].lower())
+            if category_obj is None:
+                response['errors']['name'] = 'category'
+                response['errors']['description'] = "Category is not exist"
+                return response
+            event_dict['category_id'] = category_obj.id
+
+            current_event.update_event(request, event_dict)
+
+            currnet_event_tags = EventTag.get_event_tags(request,
+                                                         self.event_id)
+            for tag in currnet_event_tags:
+                if tag not in request.validated.get('tags', []):
+                    EventTag.delete_tag(request,
+                                        currnet_event_tags[tag], self.event_id)
+            for tag in request.validated.get('tags', []):
+                tag_obj = Tag.get_by_name(request, tag.lower())
+                if tag_obj is not None:
+                    tag_id = tag_obj.id
+                    if tag not in currnet_event_tags:
+                        EventTag.add_new(request, tag_id, self.event_id)
+                else:
+                    tag_id = Tag.add_new(request, tag.lower())
+                    EventTag.add_new(request, tag_id, self.event_id)
```

Comment:
Block in lines 169-183 is a good candidate to be moved to model. Event to `Event.update_event()` method. It can accept `tags` parameter and update related tables accordingly.